File: pkg/pwrgrep/testdata/fixtures/java/EntryPoints.java
import org.junit.Test;

@RestController
public class EntryPoints {
    @GetMapping("/users/{id}")
    // ruleid: java-entry-points
    public String user(@PathVariable String id) {
        return id;
    }

    // ruleid: java-entry-points
    public static void main(String[] args) {}

    @Test
    // ruleid: java-entry-points
    public void startsUp() {}

    // ruleid: java-entry-points
    protected void doGet(HttpServletRequest req, HttpServletResponse resp) {}

    // ok: java-entry-points
    public void helper() {}
}

File: pkg/pwrgrep/testdata/fixtures/csharp/EntryPoints.cs
[ApiController]
public class UsersController : ControllerBase
{
    [HttpGet("users/{id}")]
    // ruleid: cs-entry-points
    public IActionResult Get(int id) { return Ok(); }

    // ok: cs-entry-points
    public IActionResult Helper() { return Ok(); }

    [Fact]
    // ruleid: cs-entry-points
    public void StartsUp() { }

    // ruleid: cs-entry-points
    static void Main(string[] args)
    {
        var app = WebApplication.Create(args);
        // ruleid: cs-entry-points
        app.MapGet("/hello", () => "hi");
    }
}
